### ISSUE TITLE
Add dump_last restart feature

### ISSUE THREAD

**Issue by apcraig** [CICE-Consortium/Icepack#218]
For regresssion testing, we need to write out restarts in all cases.  A handy feature would be to have a dump_last namelist setting in the Icepack driver like in CICE to turn that feature on.

Some of the regression tests are not running correctly because restarts are not being generated.

**Comment by dabail10** [comment]
I have this ready to go for icepack in an upcoming PR.

**Comment by apcraig** [comment]
OK, I just finished some mods today too.  Did you also modify the tests to turn the new feature on?  How did you decide to do that because there are some subtle issues.

**Comment by dabail10** [comment]
What I decided is that for any test where dumpfreq was set to something within the test amount of time, I turned it off. Otherwise it is on by default. I have my changes on top of the outstanding CMIP PR. Should I go ahead and issue this?

**Comment by dabail10** [comment]
No, I haven't issued the dump_last PR. If I do it before the CMIP one is merged, then it includes changes from both. I'm not sure how one should handle cumulative things like this while waiting for a merge. I will not issue it yet and take your advice to turn it on only for smoke tests that do not set dumpfreq within the length of the smoke test.

**Comment by apcraig** [comment]
It's a good question.  I think once you issue the PR and ask for reviews, then the branch should be more or less frozen (except for review issues) until the PR is pulled unless it's clear to everyone it's a work in progress.  I have created a PR in order for others to have a look at an implementation and to allow for discussion, but it's been clear it's a work in progress and I usually assign the merge to myself.  If you want to work on other features while you are waiting for a PR, then probably create a new branch.

If you want to have multiple issues dealt with in a branch, that's fine too, as long as it's doesn't get too complicated or big.  Generally, it's good to have one issue per PR unless the mods are small, independent, and clean.  But I think if you are including multiple issues, those should be included in the original PR and they should be there for the initial review.   

It's a balance between allowing the process to work cleanly and making sure everyone is productive.  It's not optimal to allow a PR to continue to creep forward with new updates while it's under review.   It's also not optimal to have to wait on a PR review and merge before taking a next step.

If your dump_last PR is on a different branch, then go ahead and issue a separate PR.  That's fine and how things are supposed to work.  If you branched off your CMIP branch, then that is a little more complicated and you'll probably want to wait for the CMIP PR to go thru, then rebase to the master on your dump_last branch.  If you are actually working on the CMIP branch and adding new modifications, then that sort of creates other issues.  I don't think there is one way to handle this, but we don't want to have PR creep where the features being addressed continue to develop on a PR once it's under review.

What does CESM say about this process?

**Comment by dabail10** [comment]
So, I started my branch dump_last on top of the CMIP branch. Hence it is cumulative. Is it better to always do new changes from master?

**Comment by apcraig** [comment]
Again, many ways to do it.  In this case, it would allow you to submit both PRs at the same time.  It depends on the changes.  If the dump_last branch depended or built on or modified some of the same changes as the CMIP branch, then it makes sense to do them sequentially.  If they are independent and without code conflicts, then it could be handy to do them separately.  It all depends.

**Comment by apcraig** [comment]
Are you OK if I create a PR for the dump_last change?

**Comment by dabail10** [comment]
No, I will do it.

**Comment by apcraig** [comment]
See PR #221
